src/MediatR/IPipelineBehavior.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace MediatR
+{
+    using System;
+    using System.Threading.Tasks;
+
+    public delegate Task<object> RequestHandlerDelegate();
+
+    public interface IPipelineBehavior
```

Comment:
I'll open a new PR with this. Making them non-generic greatly simplifies things tho.